test/util/cache/kubernetes.go

```diff
@@ -19,10 +19,12 @@ func NewFakeSharedInformerFactory(res ...runtime.Object) kubeinformers.SharedInf
 	// Start all the required informers.
 	ingressInformer := kubeInformerFactory.Extensions().V1beta1().Ingresses()
 	serviceInformer := kubeInformerFactory.Core().V1().Services()
+	secretInformer := kubeInformerFactory.Core().V1().Secrets()
 	go ingressInformer.Informer().Run(wait.NeverStop)
 	go serviceInformer.Informer().Run(wait.NeverStop)
+	go secretInformer.Informer().Run(wait.NeverStop)
 	// Wait for the caches to be synced.
-	if !kubecache.WaitForCacheSync(wait.NeverStop, ingressInformer.Informer().HasSynced, serviceInformer.Informer().HasSynced) {
+	if !kubecache.WaitForCacheSync(wait.NeverStop, ingressInformer.Informer().HasSynced, serviceInformer.Informer().HasSynced, secretInformer.Informer().HasSynced) {
```

Comment:
You can wait for all informers to be synced with `kubeInformers.WaitForCacheSync(wait.NeverStop)`

Reply:
I tried this but other tests were failing. Let's do it in a follow up PR.